src/shared/bpf-link.c

```diff
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#if HAVE_LIBBPF
+#include <bpf/libbpf.h>
+#endif
+
+#include "bpf-link.h"
+#include "serialize.h"
+
+#if HAVE_LIBBPF
+int serialize_bpf_link(FILE *f, FDSet *fds, const char *key, BPFLink *link) {
+        int fd;
+
+        assert(key);
+        assert(link);
+
+        fd = bpf_link__fd(link);
+        if (fd < 0)
+                return -errno;
+
+        return serialize_fd(f, fds, key, fd);
+}
+
+BPFLink *bpf_link_free(BPFLink *link) {
+        if (link && libbpf_get_error(link) == 0)
```

Comment:
I think this check could be omitted, `bpf_link__destroy` has the same check inside: https://github.com/libbpf/libbpf/blob/adaf538bcac57273bbb151915f30642eb3e06831/src/libbpf.c#L9400

Reply:
Yes, starting from libbpf 0.1. What's the minimum version systemd will require?

Reply:
I think we're targeting 0.2.0.